beets/importer.py

```diff
@@ -565,14 +581,27 @@ def duplicate_items(self, lib):
     def remove_duplicates(self, lib):
         duplicate_items = self.duplicate_items(lib)
         log.debug('removing {0} old duplicated items', len(duplicate_items))
+        #  try using the MB track id to compare if it exists for all tracks
+        #  else use the artist, title, and album
```

Comment:
I confess I don't quite understand this logic yet. Is the idea to only remove duplicates that have MBID matches (as long as everybody has an MBID)?

If so, I wonder if that shouldn't be a separate feature… that is, the logic that decides _what is a duplicate_ would seem to be somewhat orthogonal to deciding _what to do with those duplicates_, i.e., picking the best one for an "upgrade." Does this make sense?

Reply:
This is a part that I did a quick and dirty fix and meant to ask about. I'm not entirely sure how to compare the tracks to *know* that they're duplicates in the first place. I know that they are duplicates by this stage, but actually finding the 1:1 match is hard because there's no unique ID in the file objects.

My initial thought was to use the MB ID since that's guaranteed to be unique to a replacement song, but not all tracks have that, so the fallback is those three fields. It's clunky and if there's a simple solution that I'm missing then please share, because this isn't a great solution.

Reply:
I thought all tracks have MBID. Even if you use any other source, mbid is still populated - for example, when I use Spotify, the mbid is the Spotify track or album  id respectively.

Reply:
@arsaboo  What if you disable the autotagger?

Reply:
If you disable the autotagger (assuming that means `as-is`), will beet still prompt for duplicates? I will let @sampsyo confirm, but my feeling is we should leave those cases out as duplicates are not well defined there.

Reply:
Ah, I see. No, with `-A`, there would be no duplicate prompt.

As for how to know that objects are duplicates: I may be just super slow here, but I would have expected `self.duplicate_items(lib)` to suffice? That is, if the question is "which items are duplicates of the current item?", then the answer could be "whatever `self.duplicate_items(lib)` returns." Does that make any sense at all?